content/en/docs/tasks/traffic-management/egress/egress-gateway-tls-origination-sds/index.md

```diff
@@ -0,0 +1,717 @@
+use SDS instead of the file-mount to provision client certificates.
+
+### Generate CA and server certificates and keys
+
+For this task you can use your favorite tool to generate certificates and keys. The commands below use
+[openssl](https://man.openbsd.org/openssl.1).
+
+1.  Create a root certificate and private key to sign the certificate for your services:
+
+    {{< text bash >}}
+    $ openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=example Inc./CN=example.com' -keyout example.com.key -out example.com.crt
+    {{< /text >}}
+
+1.  Create a certificate and a private key for `my-nginx.mesh-external.svc.cluster.local`:
+
+    {{< text bash >}}
+    $ openssl req -out my-nginx.mesh-external.svc.cluster.local.csr -newkey rsa:2048 -nodes -keyout my-nginx.mesh-external.svc.cluster.local.key -subj "/CN=my-nginx.mesh-external.svc.cluster.local/O=some organization"
+    $ openssl x509 -req -days 365 -CA example.com.crt -CAkey example.com.key -set_serial 0 -in my-nginx.mesh-external.svc.cluster.local.csr -out my-nginx.mesh-external.svc.cluster.local.crt
+    {{< /text >}}
+
+### Create secrets for the client and server
+
+1. Create Kubernetes [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to hold the server's and CA certificates.
+
+    {{< text bash >}}
+    $ kubectl create -n mesh-external secret tls nginx-server-certs --key my-nginx.mesh-external.svc.cluster.local.key --cert my-nginx.mesh-external.svc.cluster.local.crt
+    $ kubectl create -n mesh-external secret generic nginx-ca-certs --from-file=example.com.crt
+    {{< /text >}}
+
+1. Create a Kubernetes Secret to hold the CA certificate used by egress gateway to originate TLS connections:
+
+    {{< text bash >}}
+    $ kubectl create secret generic client-credential-cacert --from-file=ca.crt=example.com.crt -n istio-system
+    {{< /text >}}
+
+    Note that the secret name for an Istio CA-only certificate must end with `-cacert` and the secret **must** be 
+    created in the same namespace as Istio is deployed in, `istio-system` in this case.
+    
+    {{< warning >}}
+    The secret name **should not** begin with `istio` or `prometheus`, and the secret **should not** contain a `token` field. 
+    {{< /warning >}}
+
+### Deploy a simple TLS server
+
+To simulate an actual external service that supports the simple TLS protocol,
+deploy an [NGINX](https://www.nginx.com) server in your Kubernetes cluster, but running outside of
+the Istio service mesh, i.e., in a namespace without Istio sidecar proxy injection enabled.
+
+1.  Create a namespace to represent services outside the Istio mesh, namely `mesh-external`. Note that the sidecar proxy will
+    not be automatically injected into the pods in this namespace since the automatic sidecar injection was not
+    [enabled](/docs/setup/additional-setup/sidecar-injection/#deploying-an-app) on it.
+
+    {{< text bash >}}
+    $ kubectl create namespace mesh-external
+    {{< /text >}}
+
```

Comment:
```suggestion

1. Create Kubernetes [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to hold the server's and CA certificates.

    {{< text bash >}}
    $ kubectl create -n mesh-external secret tls nginx-server-certs --key my-nginx.mesh-external.svc.cluster.local.key --cert my-nginx.mesh-external.svc.cluster.local.crt
    $ kubectl create -n mesh-external secret generic nginx-ca-certs --from-file=example.com.crt
    {{< /text >}}

```

Reply:
namespace needs to be created before secrets can be created, I moved that step to before secret creation but under the same section